### PR TITLE
Fix(tsql)!: parse TININT into UTINYINT to improve transpilation

### PR DESCRIPTION
**Pull request by georgesittas**
Fixes #3469

### CODE REVIEW

File: sqlglot/dialects/tsql.py [hunk 1]
```diff
@@ -526,6 +526,12 @@ class Parser(parser.Parser):
             *parser.Parser.TYPE_TOKENS,
         }
 
+        TYPE_CONVERTER = {
+            # TINYINT is a 1-byte unsigned integer
+            # https://learn.microsoft.com/en-us/sql/t-sql/data-types/int-bigint-smallint-and-tinyint-transact-sql?view=sql-server-ver16
+            exp.DataType.Type.TINYINT: lambda _: exp.DataType(this=exp.DataType.Type.UTINYINT),
```
Comment (tobymao) [inline sqlglot/dialects/tsql.py:532]: why do you need the type_converter, can't we use the type mapping?

Comment (georgesittas) [reply sqlglot/dialects/tsql.py:532]: Which type mapping? Tokenizer?

Comment (tobymao) [reply sqlglot/dialects/tsql.py:532]: yea just map the tinyint token for tsql to uint

Comment (georgesittas) [reply sqlglot/dialects/tsql.py]: done

**Pull request merged**